packages/checkout/widgets-lib/src/lib/hooks/useBalance.ts

```diff
@@ -0,0 +1,78 @@
+import { useCallback, useEffect, useState } from 'react';
+import { Checkout, GetBalanceResult } from '@imtbl/checkout-sdk';
+import { Web3Provider } from '@ethersproject/providers';
+import { getTokenBalances } from '../../widgets/wallet/functions/tokenBalances';
+import { DEFAULT_BALANCE_RETRY_POLICY } from '../constants';
+import { useInterval } from './useInterval';
+
+const REFRESH_BALANCE_INTERVAL_MS = 10000;
+
+export interface UseBalanceParams {
+  checkout: Checkout | undefined;
+  provider: Web3Provider | undefined,
+  refreshCallback: (balance: GetBalanceResult[]) => void;
+  errorCallback: (error: Error) => void;
+}
+
+export const useBalance = ({
+  checkout,
+  provider,
+  refreshCallback,
+  errorCallback,
+}: UseBalanceParams) => {
+  const [balancesLoading, setBalancesLoading] = useState(true);
+
+  const refreshBalances = useCallback(async (
+    // If silent is true, the balances will not be set to loading
+    silent: boolean = false,
+  ) => {
+    if (!checkout || !provider) return;
+    try {
+      const network = await checkout.getNetworkInfo({
```

Comment:
we should have the `chainId` as a prop and if present skip this call otherwise we would have an RPC call every time we want to refresh the balances

Reply:
Hmmm I don't see those RPC calls every balance refresh. I'm seeing that the `getNetworkInfo` calls checks the `network` object on the provider first before doing a call. So I think it must be using cached network data for that provider instance... if the chain changes the widgets listen into that window event and rewraps the provider anyway causing a refresh, but we are assuming that is all going to be managed upstream if I remove these checks.

Reply:
Im not seeing any rpc calls relating to this call in particular (not sure what the provider is doing around caching or whatever because I thought the call in the sdk `await web3Provider.getNetwork();` does make an rpc call)

However Im noticing a lot of rpc spam particularly when the chain id is sepolia. Just hard coded it like this:
`const balances: GetBalanceResult[] = await getTokenBalances(checkout, provider, ChainId.SEPOLIA);`
Do not see the same thing when I set to zkevm

Reply:
Oh interesting, I'm also seeing those `eth_chainId` calls but only for Sepolia.
<img width="473" alt="Screenshot 2024-01-25 at 1 51 35 pm" src="https://github.com/immutable/ts-immutable-sdk/assets/1617003/cb57801f-0b82-492d-bf2f-a9e95906b3ec">

Reply:
I can see the `getAllBalances` is checking for supported chains anyway.. so happy to remove this as you suggested @andrearampin .. I think if the underlying provider changes it still has checks in place and if it still goes through and is a problem the window events will intercept it anyway.
```  if (!config.networkMap.get(chainId)) {
    throw new CheckoutError(
      `chain ID ${chainId} not supported by the environment`,
      CheckoutErrorType.CHAIN_NOT_SUPPORTED_ERROR,
    );
  }```

Reply:
Number of requests are probably due to https://github.com/immutable/ts-immutable-sdk/blob/main/packages/checkout/widgets-lib/src/widgets/wallet/views/WalletBalances.tsx#L127 -- let remove that logic, it has really little use